src/sys/socket/mod.rs

```diff
@@ -655,6 +728,11 @@ impl<'a> ControlMessage<'a> {
             ControlMessage::ScmCredentials(creds) => {
                 &creds.0 as *const libc::ucred as *const u8
             }
+            #[cfg(any(target_os = "freebsd", target_os = "dragonfly"))]
+            ControlMessage::ScmCreds => {
+                // The kernel overwrites the data, don't worry
+                return
```

Comment:
Could this result in `cmsg_data` being uninitialized?  That way might lead to UB.  I think that you should zero it instead.